### DISCUSSION TITLE
"u$ called [setState] during rendering, which you should avoid" warnings

### DISCUSSION THREAD

**Discussion by luber** [SVG-Edit/svgedit#876] [General]
Hi,
I have embedded SVG Editor (v7.3.0) into an angular application. It works fine when I run an angular app in development mode.
But when I build it in prod mode (with optimization: true) - I can see the next warning messages in the browser console:
"u$ called [setState] during rendering, which you should avoid."
<img width="603" alt="image" src="https://user-images.githubusercontent.com/3791692/214107046-bb68c9ce-2469-4a4a-b224-81d1af2e723d.png">

And when I try to edit some shapes in the editor - I can see that some toolbar controls (se-spin-input) do not render `input` HTML elements:
<img width="957" alt="image" src="https://user-images.githubusercontent.com/3791692/214107621-b7f868f5-7e35-4767-913f-2c954c19d304.png">

When running in the angular app without optimization (in develop mode), there are no such messages in the browser console and controls are rendered correctly...

I would appreciate any advice on how to deal with this.

**Comment by jfhenon** [comment]
We had a few issues with the elix library (underlying web component library). This is why a number of changes were made to remove some elix components. I never experimented the symptom you mention (but this was never tested with Angular). 
I'm not blaming elix as it may be the way svgedit is implementing their component library. 
However, to fix your issue, you need to if we do something strange in the src/editor/components/seSpinInput.js or rewrite this component...

**Reply by luber** [reply]
I am not sure what was the root cause of that issue, but after spending a day with trials and errors, when I built svgedit npm package manually and referenced it to my angular project - the issue was gone...
